oximeter/collector/src/agent.rs

```diff
@@ -969,17 +969,24 @@ mod tests {
             .unwrap()
             .statistics()
             .await;
-        let count = stats
+
+        // The collections _should_ always fail due to a 500, but it's possible
+        // that we also get collections failing because there's already one in
+        // progress. See
+        // https://github.com/oxidecomputer/omicron/issues/7255#issuecomment-2711537164
+        // for example.
+        //
+        // Sum over all reasons to get the correct count.
+        dbg!(&stats.failed_collections);
+        let count: usize = stats
             .failed_collections
-            .get(&FailureReason::Other(
-                reqwest::StatusCode::INTERNAL_SERVER_ERROR,
```

Comment:
Do you want to keep a check for this code specifically? As written I think this could pass if we got a failure _only_ due to "collection in progress", although maybe that's fine?

Reply:
I'm not sure how to write it with anything other than comparing the total number of failed collections. I want to test that they're all 500s, but that's obviously flaky. I don't know where to draw a line that won't land us back here later.

Reply:
Hmm, what if we matched explicitly? Untested but something like

```rust
let count: usize = stats
    .failed_collections
    .iter()
    .filter_map(|(reason, value)| match reason {
        FailureReason::Other(INTERNAL_SERVER_ERROR) => Some(value.datum.value() as usize),
        FailureReason::CollectionsInProgress => None,
        _ => panic!("unexpected failure reason: {reason:?}"),
    })
    .sum();
```

Reply:
Oh, I see what you meant. Yes, these should be the only two reasons we get failures. I _think_...but I've been wrong before.

Reply:
Small fixup in deb6b68